src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -148,38 +177,55 @@ private void CheckDisposed()
             }
         }
 
-        private void WriteInternal(byte[] buffer, int offset, int count)
+        private async Task InternalWriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
         {
             if (_chunkedMode)
             {
                 string chunkSizeString = String.Format("{0:x}\r\n", count);
                 byte[] chunkSize = Encoding.UTF8.GetBytes(chunkSizeString);
 
-                WriteData(chunkSize, 0, chunkSize.Length);
+                await InternalWriteDataAsync(chunkSize, 0, chunkSize.Length, token);
 
-                WriteData(buffer, offset, count);
-                WriteData(s_crLfTerminator, 0, s_crLfTerminator.Length);
+                await InternalWriteDataAsync(buffer, offset, count, token);
+                await InternalWriteDataAsync(s_crLfTerminator, 0, s_crLfTerminator.Length, token);
             }
             else
             {
-                WriteData(buffer, offset, count);
+                await InternalWriteDataAsync(buffer, offset, count, token);
```

Comment:
This might be unnecessary microoptimization, but this could be made more efficent by doing:
```C#
private Task InternalWriteAsync(byte[] buffer, int offset, int count, CancellationToken token)
{
    return _chunkedMode ?
        InternalWriteChunkedModeAsync(buffer, offset, count, token) :
        InternalWriteDataAsync(buffer, offset, count, token);
}

private async Task InternalWriteChunkedModeAsync(byte[] buffer, int offset, int count, CancellationToken token)
{
    Debug.Assert(_chunkedMode);
    ... // the contents of your "if (_chunkedMode)" branch
}
```
That way, when !_chunkedMode, there's no additional async method invocation.